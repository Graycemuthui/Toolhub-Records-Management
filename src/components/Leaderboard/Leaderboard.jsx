import { records } from "../data/Mockdata";
import Container from "react-bootstrap/Container";
import Table from "react-bootstrap/Table";

const Leaderboard = () => {
  const filters = records.sort((a, b) => {
    if (a.edits > b.edits) {
      return -1; //{1, 2, 3, 4, 4}
    } else if (a.edits < b.edits) {
      return 1;
    } else return 0;
  });

  const nonZeroFilters = filters.filter((filter) => filter.edits > 0);
  return (
    <Container className="p-5">
      <Table striped bordered hover variant="dark" size="sm">
        <thead>
          <tr>
            <th>Name</th>
            <th>Edits</th>
          </tr>
        </thead>
        <tbody>
          {nonZeroFilters.map(({ id, name, edits }) => (
            <tr key={id}>
              <td>
                <p>{name}</p>
              </td>
              <td>
                <p>{edits}</p>
              </td>
            </tr>
          ))}
        </tbody>
      </Table>
    </Container>
  );
};

export default Leaderboard;
