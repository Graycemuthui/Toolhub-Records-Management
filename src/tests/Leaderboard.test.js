import React from "react";
import TestRenderer from "react-test-renderer";
import Leaderboard from "../components/Leaderboard/Leaderboard";

it("Leaderboard render correctly", () => {
  const tree = TestRenderer.create(<Leaderboard />).toJSON();
  expect(tree).toMatchSnapshot();
});
