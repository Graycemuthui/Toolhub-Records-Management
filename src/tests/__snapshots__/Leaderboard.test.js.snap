// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Leaderboard render correctly 1`] = `
<div
  className="p-5 container"
>
  <table
    className="table table-dark table-sm table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Edits
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          <p>
            Bright Kweku
          </p>
        </td>
        <td>
          <p>
            34
          </p>
        </td>
      </tr>
      <tr>
        <td>
          <p>
            Ever Melisandre
          </p>
        </td>
        <td>
          <p>
            29
          </p>
        </td>
      </tr>
      <tr>
        <td>
          <p>
            Grayce Muthui
          </p>
        </td>
        <td>
          <p>
            24
          </p>
        </td>
      </tr>
      <tr>
        <td>
          <p>
            Shaq Deff
          </p>
        </td>
        <td>
          <p>
            24
          </p>
        </td>
      </tr>
      <tr>
        <td>
          <p>
            Kiana Holcomb
          </p>
        </td>
        <td>
          <p>
            14
          </p>
        </td>
      </tr>
      <tr>
        <td>
          <p>
            Kamau Moii
          </p>
        </td>
        <td>
          <p>
            4
          </p>
        </td>
      </tr>
      <tr>
        <td>
          <p>
            Kofi Bright
          </p>
        </td>
        <td>
          <p>
            4
          </p>
        </td>
      </tr>
      <tr>
        <td>
          <p>
            Jaime Lannister
          </p>
        </td>
        <td>
          <p>
            3
          </p>
        </td>
      </tr>
      <tr>
        <td>
          <p>
            Daenerys Targaryen
          </p>
        </td>
        <td>
          <p>
            2
          </p>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
